modules/core/src/main/java/org/apache/ignite/internal/binary/streams/BinaryAbstractOutputStream.java

```diff
@@ -29,14 +29,17 @@ public abstract class BinaryAbstractOutputStream extends BinaryAbstractStream
     /** Minimal capacity when it is reasonable to start doubling resize. */
     private static final int MIN_CAP = 256;
 
+    /** Max capacity when it is still reasonable doubling resize. */
+    protected static final int MAX_CAP = 32 * 1024 * 1024;
```

Comment:
This changes looks like overcomplication without clear performance benefits.
Do we really need them? 
Can we revert it here and move to the separate PR only if there are clear real-worls benchmarks that show some advantage of new restriction